analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ConditionalsWithSameCondition.CSharp9.cs

```diff
@@ -52,17 +52,57 @@ public void Foo()
                 f = cond ? new Apple() : new Orange();
             }
 
+            if (f is null) { }
+            if (f is null) { }  // Noncompliant
+
+            char ch = 'x';
+            if (ch is >= 'a' and <= 'z') { }
+            if (ch is >= 'a' and <= 'z') { } // Noncompliant
+            if (ch is <= 'z' and >= 'a') { } // FN, rule is syntax based. Only null check equivalence is currenty supported in CSharpEquivalenceChecker
+
+            if (ch is (>= 'a' and <= 'z') or (>= 'A' and <= 'Z') or '.' or '?') { }
+            if (ch is (>= 'a' and <= 'z') or (>= 'A' and <= 'Z') or '.' or '?') { } // Noncompliant
+            if (ch is (>= 'a' and <= 'z') or (>= 'A' and <= 'Z') or '?' or '.') { } // FN, rule is syntax based. Only null check equivalence is currenty supported in CSharpEquivalenceChecker
+            if (ch is (>= 'A' and <= 'Z') or (>= 'a' and <= 'z') or '?' or '.') { } // FN, rule is syntax based. Only null check equivalence is currenty supported in CSharpEquivalenceChecker
+
             if (f is not null) { }
-            if (f != null) { } // FN, same as above
+            if (f is not not not null) { }  // Noncompliant, same meaning
+            if (f != null) { }              // Noncompliant, same meaning
 
             if (f is null) { }
-            if (f == null) { } // FN, same as above
+            if (f == null) { }          // Noncompliant, same meaning
+            if (!(f is not null)) { }   // Noncompliant, same meaning
+
+            if (!(f is null)) { }
+            if (f != null) { }          // Noncompliant, same meaning
+            if (f is not not null) { }  // Compliant, equal to f == null
+
+            if (f is Apple) { }
+            if (f is Apple) { }   // Noncompliant
+
+            if (f is Apple or Orange) { }
+            if (f is Apple or Orange) { }   // Noncompliant
+            if (f is Orange or Apple) { }   // FN, rule is syntax based. Only null check equivalence is currenty supported in CSharpEquivalenceChecker
+
+            if (f is not Apple) { }
+            if (f is not Apple) { } // Noncompliant
+
+            if (f is { Size: >= 5, Value: 0 }) { }
+            if (f is { Size: >= 5, Value: 0 }) { } // Noncompliant
+            if (f is { Value: 0, Size: >= 5 }) { } // FN, rule is syntax based. Only null check equivalence is currenty supported in CSharpEquivalenceChecker
+
+            if (f is { Size: >= 5 }) { }
+            if (f is not { Size: < 5 }) { } // FN, rule is syntax based. Only null check equivalence is currenty supported in CSharpEquivalenceChecker
```

Comment:
I finally created a compilable snippet with positional patterns that you could add to the tests - [see sharplab](https://sharplab.io/#v2:EYLgtghglgdgNAFxBAzguATEBqAPgAQGYACfAJmIGFiBvAWAChjnST8AWYgWQAoBBAA4CANgFNiEAJS1GLOcSgAzYjwgKUKgEQBlAO6jRCTXGKaASqIybJk2fOb0m94gF879pSrVQNPHfsNjUwsrG3d5R2dXcLlPVXVaABVUBFEQUz0DIxNKAHthXIAndPNLTRdbJ3tI5zcq6IY6xiJSCkERcRrmFvwARgAGYmS0UQBucJ6Bqnyi8fqezgARUQBjXJg0QoBXFYQeXK2EUimADxMDo77BgE9pAF4APhUz4lviO5Vh1JyZwsk5lxAA)

```
public class C {
    public void M(Apple a) {
        if (a is ("Sweet", "Red"))
        {
        }
        if (a is ("Sweet", "Red"))
        {
        }
        if (a is {Taste: "Sweet", Color: "Red"})
        {
        }
    }
}

public class Apple {
    public string Taste;
    public string Color;
    public void Deconstruct(out string x, out string y) => (x, y) = (Taste, Color);
}
```

(sorry for the late comment)